tests/unittests/core/worker/test_experiment.py

```diff
@@ -641,6 +641,20 @@ def test_status_is_pending_when_increase_max_trials(self, exp_config):
 
         assert not exp.is_done
 
+    def test_no_increment_when_child_exist(self, create_db_instance):
+        """Check that experiment is not incremented when asked for v1 while v2 exists."""
+        query = dict(name='experiment_test', metadata={'user': 'corneauf'}, version=1)
+        get_storage().create_experiment(query)
+
+        query['version'] = 2
+        query.pop('_id')
+
+        get_storage().create_experiment(query)
+
+        exp = Experiment('experiment_test', version=1)
+
+        assert exp.version == 1
+
```

Comment:
The problem was when we do experiment.configure, where we would create a new Experiment for which we didn't pass self.version.